### PR TITLE
Add note regarding Windows VOLUME limitations

### PR DESCRIPTION
**Pull request by johnmaguire**
**- What I did**

I added a note regarding some limitations of Docker volumes on Windows to the builder documentation under the `VOLUME` header, per #20127.

**- How I did it**

With the uncontested, greatest editor of all time, vim.

**- How to verify it**

N/A

**- Description for the changelog**

N/A

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: docs/reference/builder.md [hunk 1]
```diff
@@ -1222,6 +1222,12 @@ This Dockerfile results in an image that causes `docker run`, to
 create a new mount point at `/myvol` and copy the  `greeting` file
 into the newly created volume.
 
+> **Warning**:
+> When using Windows-based containers, the destination of a volume or
+> bind-mount inside the container must be one of: a non-existing or empty
+> directory; or a drive other than C:. Further, the source of a bind mount must
+> be a local directory, not a file.
+
```
Comment (JohnMaguire) [inline docs/reference/builder.md]: I ripped this text off of here: https://github.com/docker/docker/blob/master/docs/reference/commandline/run.md#mount-volume--v---read-only

Original read:

> The following examples will fail when using Windows-based containers, as the destination of a volume or bind-mount inside the container must be one of: a non-existing or empty directory; or a drive other than C:. Further, the source of a bind mount must be a local directory, not a file.

File: docs/reference/builder.md [hunk 2]
```diff
@@ -1222,6 +1222,12 @@ This Dockerfile results in an image that causes `docker run`, to
 create a new mount point at `/myvol` and copy the  `greeting` file
 into the newly created volume.
 
+> **Warning**:
+> When using Windows-based containers, the destination of a volume or
+> bind-mount inside the container must be one of: a non-existing or empty
```
Comment (thaJeztah) [inline docs/reference/builder.md]: Can you remove `or bind-mount` here? Bind mounts are not used in Dockerfiles

File: docs/reference/builder.md [hunk 3]
```diff
@@ -1222,6 +1222,12 @@ This Dockerfile results in an image that causes `docker run`, to
...
+> directory; or a drive other than C:. Further, the source of a bind mount must
```
Comment (thaJeztah) [inline docs/reference/builder.md]: Same here; can you remove "Further, the source of a bind mount must be a local directory, not a file." ?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks for working on this! I left some comments, could you make those changes and amend your commit?

Also ping @jhowardmsft @johnstep for review

**Comment by JohnMaguire** [comment]
@thaJeztah Thanks for the comments, updated!

**Review by thaJeztah — APPROVED** [review]
perfect! LGTM

**Comment by jhowardmsft** [comment]
Text is good. Just questioning whether it's really a `warning`. Should it not be a `note`?

**Comment by thaJeztah** [comment]
"note" works for me

**Comment by JohnMaguire** [comment]
👍  Updated.

**Comment by jhowardmsft** [comment]
LGTM. Thanks @JohnMaguire. 👍

**Pull request merged**